### PR TITLE
chore(mui): migrate to Material-ui v5

### PR DESCRIPTION
**Pull request by bdauria**
## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie
- [x] 22.04.x (master)

### CODE REVIEW

File: www/front_src/src/Extensions/ExtensionDetailsPopup/LoadingSkeleton.tsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import * as React from 'react';
 
-import { useTheme, makeStyles } from '@material-ui/core';
-import { Skeleton, SkeletonProps } from '@material-ui/lab';
+import { useTheme, Skeleton, SkeletonProps } from '@mui/material';
+import makeStyles from '@mui/styles/makeStyles';
+// import { SkeletonProps } from '@mui/lab';
```
Comment (bdauria) [inline www/front_src/src/Extensions/ExtensionDetailsPopup/LoadingSkeleton.tsx:5]:
```suggestion
```

File: www/front_src/src/Provider/index.test.tsx [hunk 2]
```diff
@@ -69,6 +69,7 @@ const retrievedTranslations = {
 
 jest.mock('../App', () => {
   const ComponentWithUserContext = (): JSX.Element => {
+    // eslint-disable-next-line react/jsx-no-useless-fragment
```
Comment (Thebarda) [inline www/front_src/src/Provider/index.test.tsx:72]: We can replace it by an empty div tag

File: www/front_src/src/Resources/Actions/Resource/Downtime/Dialog.tsx [hunk 3]
```diff
@@ -63,26 +51,22 @@ interface Props {
...
-  ...pickerCommonProps,
-  format: 'LT',
-} as Omit<TimePickerProps, 'onChange' | 'value'>;
+const renderDateTimePickerTextField =
+  (ariaLabel: string) =>
+  ({ inputRef, inputProps, InputProps }: TextFieldProps): JSX.Element => {
+    return (
+      <TextField
+        // eslint-disable-next-line react/no-unstable-nested-components
+        EndAdornment={(): JSX.Element => <div>{InputProps?.endAdornment}</div>}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Actions/Resource/Downtime/Dialog.tsx:60]:
```suggestion
        EndAdornment={(): JSX.Element => <div>{InputProps?.endAdornment || ''}</div>}
```
Does this work ?

File: www/front_src/src/Resources/Filter/Edit/index.test.tsx [hunk 4]
```diff
@@ -213,8 +209,8 @@ describe(EditFilterPanel, () => {
 
     await makeDnd({
       direction: DND_DIRECTION_DOWN,
-      getByText,
-      getDragEl: () => firstFilterDraggable,
+      // getByText,
```
Comment (Thebarda) [inline www/front_src/src/Resources/Filter/Edit/index.test.tsx:212]:
```suggestion
```

File: www/front_src/src/Resources/Graph/Performance/Graph/Annotations/EventAnnotations.tsx [hunk 5]
```diff
@@ -29,7 +29,7 @@ const EventAnnotations = ({
   color,
   resourceId,
 }: Props): JSX.Element => {
-  const events = filter<TimelineEvent>(propEq('type', type), timeline);
+  const events = filter(propEq('type', type), timeline);
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/Graph/Annotations/EventAnnotations.tsx:32]: Using `@types/ramda v0.27.45`, the typing `<TimelineEvent>` should work

File: www/front_src/src/Resources/Graph/Performance/Graph/index.tsx [hunk 6]
```diff
@@ -560,6 +560,7 @@ const GraphContent = ({
             />
           </Group.Group>
           <TimeShiftContext.Provider
+            // eslint-disable-next-line react/jsx-no-constructed-context-values
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/Graph/index.tsx:563]: As suggested by the [rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-constructed-context-values.md), we should wrap the context values inside a useMemo

File: www/front_src/src/Resources/Graph/Performance/TimePeriods/DateTimePickerInput.tsx [hunk 7]
```diff
@@ -1,52 +1,69 @@
...
+      maxDate={dayjs(maxDate)}
+      minDate={dayjs(minDate)}
+      open={isOpen}
+      renderInput={renderDateTimePickerTextField(toggleIsOpen)}
       value={date}
-      variant="inline"
-      onChange={(value): void => setDate(new Date(value?.toDate() || 0))}
+      onChange={(value): void => {
+        setDate(new Date(value?.toDate() || 0));
+      }}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/TimePeriods/DateTimePickerInput.tsx:66]: No keyboard support on these DateTimePicker ?

File: www/front_src/src/Resources/Listing/columns/Graph.tsx [hunk 8]
```diff
@@ -103,9 +104,11 @@ const GraphColumn = ({
     return (
       <IconColumn>
         <HoverChip
+          // eslint-disable-next-line react/no-unstable-nested-components
```
Comment (Thebarda) [inline www/front_src/src/Resources/Listing/columns/Graph.tsx:107]: I guess you need to extract the Component to remove this eslint error

File: www/front_src/src/Resources/Listing/useLoadResources/index.test.tsx [hunk 9]
```diff
@@ -36,6 +36,7 @@ let context: ResourceContext;
 const LoadResourcesComponent = (): JSX.Element => {
   useLoadResources();
 
+  // eslint-disable-next-line react/jsx-no-useless-fragment
```
Comment (Thebarda) [inline www/front_src/src/Resources/Listing/useLoadResources/index.test.tsx:39]: You can replace Fragment by an empty div

File: www/front_src/src/Resources/Graph/Performance/Graph/Annotations/EventAnnotations.tsx (hunk 5 above)
Comment (bdauria) [reply www/front_src/src/Resources/Graph/Performance/Graph/Annotations/EventAnnotations.tsx:32]: It's not needed though as the type is properly infered without the generic

File: www/front_src/src/Resources/Graph/Performance/TimePeriods/DateTimePickerInput.tsx (hunk 7 above)
Comment (bdauria) [reply www/front_src/src/Resources/Graph/Performance/TimePeriods/DateTimePickerInput.tsx:66]: Not at this time, we can surely add this support in a later stage

**Comment by Thebarda** [comment]
Screenshot 2021-12-28 at 12 13 44

Those buttons looks strange. I guess, you have to change the color to the default

**Review by Shonan10 — DISMISSED** [review]

**Review by Jordan91 — DISMISSED** [review]

**Review by Thebarda — DISMISSED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by Shonan10 — APPROVED** [review]

**Review by Jordan91 — APPROVED** [review]

**Pull request merged**